src/compiler/checker.ts

```diff
@@ -5750,6 +5755,26 @@ namespace ts {
                 }
                 return Ternary.False;
             }
+
+            function enumRelatedTo(source: Type, target: Type) {
+                if (source.symbol.name !== target.symbol.name ||
+                    source.symbol.flags & SymbolFlags.ConstEnum ||
+                    target.symbol.flags & SymbolFlags.ConstEnum) {
+                    return Ternary.False;
+                }
+                const sourceMembers = resolveStructuredTypeMembers(getTypeOfSymbol(source.symbol)).properties;
+                const targetMembers = resolveStructuredTypeMembers(getTypeOfSymbol(target.symbol)).properties;
+                const targetNames = arrayToMap(targetMembers, member => member.name);
+                for (const member of sourceMembers) {

```

Comment:
you still have to filter list to use only elements with `SymbolFlags.EnumMember` otherwise you'll pick members from merged module declarations which is presumably not desirable.
```ts
enum A { X }
module A {
    export let Y = () => 42;
}
A.$ // A has both X and Y

```